vars/signArtifacts.groovy

```diff
@@ -15,15 +15,16 @@ SignArtifacts signs the given artifacts and saves the signature in the same dire
 @param Map[platform] <Required> - The distribution platform for signing.
 */
 void call(Map args = [:]) {
-
     if (args.sigtype.equals('.rpm')) {
-        echo "RPM Add Sign"
+        withCredentials([string(credentialsId: 'jenkins-rpm-signing-props', variable: 'configs')]) {
+                def props = readJSON(text: configs)
+                def signingAccount = props['account']
+                def signingPassphraseSecretsArn = props['passphrase_secrets_arn']
+                def signingSecretKeyIdSecretsArn = props['secret_key_id_secrets_arn']
+
+            echo 'RPM Add Sign'
 
-        withAWS(role: "${SIGN_ASM_ROLE}", roleAccount: "${SIGN_ASM_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-signing-session') {
-            withCredentials([
-                string(credentialsId: 'jenkins-rpm-signing-asm-pass-id', variable: 'SIGNING_PASS_ID'),
-                string(credentialsId: 'jenkins-rpm-signing-asm-secret-id', variable: 'SIGNING_SECRET_ID')])
-                {
+            withAWS(role: 'jenki-jenki-asm-assume-role', roleAccount: "${signingAccount}", duration: 900, roleSessionName: 'jenkins-signing-session') {
```

Comment:
This role is still env var?

Reply:
Nope its an actual role name. No need to keep role names as secrets or env variables.